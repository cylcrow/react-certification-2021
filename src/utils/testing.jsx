import React from "react";
import { render } from '@testing-library/react';
import { ThemeProvider } from "styled-components";
import { lightTheme } from '../providers/theme/themes';

export const renderWithTheme = (Component, theme = lightTheme) => {
    return render(
        <ThemeProvider theme={{ theme, switchTheme: jest.fn() }}>
        { Component }
        </ThemeProvider>
    );
}