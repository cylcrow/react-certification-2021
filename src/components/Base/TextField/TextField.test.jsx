import React from 'react';
import '../../../utils/testing';
import 'jest-styled-components';
import { render, unmountComponentAtNode } from 'react-dom';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import TextField from './TextField';

let container;
const build = () => {
  render(<TextField />, container);
};

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
  return container;
});

describe('TextField', () => {
  it('renders', () => {
    const wrapper = build(<TextField />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('TextField styles', () => {
  it('applies default styling', () => {
    const tree = mount(<TextField />);
    expect(tree).toHaveStyleRule('background', '#26a69a');
    expect(tree).toHaveStyleRule('color', 'white');
    expect(tree).toHaveStyleRule('border-radius', '4px');
    expect(tree).toHaveStyleRule('padding', '8px');
    expect(tree).toHaveStyleRule('border', '0');
  });
});
