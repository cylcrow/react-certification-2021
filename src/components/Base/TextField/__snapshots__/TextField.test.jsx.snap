// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField renders 1`] = `undefined`;
